client/client_update_test.go

```diff
@@ -893,6 +965,53 @@ func TestUpdateNonRootRemoteCorruptedNoLocalCache(t *testing.T) {
 				role: role,
 			}, testData, true)
 		}
+
+		switch role {
+		case data.CanonicalSnapshotRole:
+			for _, missing := range []string{data.CanonicalRootRole, data.CanonicalTargetsRole} {
+				testData := swizzleExpectations{
+					desc: fmt.Sprintf("snapshot missing root meta checksum"),
+					// TODO: this should probably be always client.ErrMissingMeta
```

Comment:
That would be awesome, thanks!  I think some of this might eventually be obsoleted if we validate the metadata beforehand, but definitely we should

- check if the file checksum metadata on the snapshot before using it (I think we do that for downloading the snapshots and maybe the targets?)
- also fail if the sha256 checksum isn't present

If you could help with that that would be awesome!  Either here or in another PR would work.